File: 2º/PDOO/ProyectoJAVA/src/civitas/Casilla.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author nikitastetskiy
 */
public class Casilla {
    
    private String nombre;
    
    Casilla(String nombre){                             // descanso
        this.nombre = nombre;
    }

    public String getNombre() {
        return nombre;
    }
    
    void informe (int actual, ArrayList<Jugador> todos){
        String e="El jugador "+todos.get(actual).getNombre()+" esta en la casilla "+toString();
        Diario.getInstance().ocurreEvento(e);
    }
  
    
    public boolean jugadorCorrecto (int actual, ArrayList<Jugador> todos){
        return actual<todos.size() && actual>=0;
    }
    
    void recibeJugador (int actual, ArrayList<Jugador> todos){
        informe(actual, todos);
    }
    
    @Override
    public String toString() {
        String e = "\n-> Casilla (" + nombre + ")";
        return e;
    }
    
}

File: 2º/PDOO/ProyectoRUBY/lib/casilla.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module  Civitas
  class Casilla
    def initialize(nombre)
      @nombre=nombre  
    end
    attr_reader :nombre
    def  to_s
      devolver="\nCasilla{\n"
      devolver+="\tNombre: #{@nombre}."
      return devolver;
    end
    def jugadorCorrecto(actual, todos)
      if(actual<=(todos.size)) then
        return true
      else
        return false
      end
    end
    def jugadorCorrecto(actual, todos)
      if(actual<=(todos.size)) then
        return true
      else
        return false
      end
    end
    def recibeJugador(actual, todos)
      informe(actual, todos)
    end
    def informe(actual, todos)
      Diario.instance.ocurre_evento("El jugador #{todos[actual].nombre} a caido en la casilla #{@nombre}");
    end 
  end
end
